### PR TITLE
Fix (BX-357): Fix negative confirm bid timers

### PR DESCRIPTION
**Pull request by annacarey**
This PR resolves [BX-357](https://artsyproduct.atlassian.net/browse/BID-357)

### Description

During QA, we discovered that the app has additional surfaces that use a timer component (on the confirm id modal). That timer component was always receiving the sale's `endAt` date even if the sale had cascading end times turned on. The result was that the the timer countdown showed a negative number when a user placed bids as lots were closing because the sale level end date had passed, but the lots were still open. The fix is to pass in the sale artwork's `endAt` date to the timer if the sale has cascading end times so the countdown will show time left on the lot, not on the sale.

The results: 

Screen Shot 2022-04-20 at 3 27 29 PM

Screen Shot 2022-04-20 at 3 27 43 PM

### PR Checklist

- [ ] I tested my changes on **iOS** / **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look on our [docs], or get in touch with us.

[app state migration]: /docs/adding_state_migrations.md
[feature flag]: /docs/developing_a_feature.md
[docs]: /docs/README.md

### CODE REVIEW

File: src/app/Components/Bidding/Screens/BidResult.tests.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { BidResult_sale_artwork } from "__generated__/BidResult_sale_artwork.graphql"
 import { dismissModal, navigate } from "app/navigation/navigate"
 import { __globalStoreTestUtils__ } from "app/store/GlobalStore"
-import { renderWithWrappers } from "app/tests/renderWithWrappers"
+import { renderWithWrappers, renderWithWrappersTL } from "app/tests/renderWithWrappers"
```
Comment (olerichter00) [inline src/app/Components/Bidding/Screens/BidResult.tests.tsx:4]: 👏

**Review by olerichter00 — APPROVED** [review]
Looks good 🌟

File: src/app/Components/Bidding/Screens/BidResult.tests.tsx [hunk 2]
```diff
@@ -195,6 +195,42 @@ describe("BidResult component", () => {
...
+        const { getByText } = renderWithWrappersTL(<BidResult {...propsForCascadingSale} />)
+        // Today is May 10. Sale artwork's end time is May 13. Sale's end day is May 10.
+
+        const timerText = getByText("03d 00h 00m 10s")
+        expect(timerText).toBeTruthy()
+      })
+
+      it("shows the sale's end time if the sale does not have cascading end times", () => {
+        const { getByText } = renderWithWrappersTL(<BidResult {...propsForNonCascadingSale} />)
+        // Today is May 10. Sale artwork's end time is May 13. Sale's end day is May 10.
```
Comment (nspinazz89) [inline src/app/Components/Bidding/Screens/BidResult.tests.tsx:216]: not sure the text of this comment is correct, shouldn't it say the sale artwork's end time is may 10th as well?

File: src/app/Components/Bidding/Screens/BidResult.tsx [hunk 3]
```diff
@@ -17,7 +17,7 @@ import { BidderPositionResult } from "../types"
 
 import { BidResult_sale_artwork } from "__generated__/BidResult_sale_artwork.graphql"
 import { FancyModalHeader } from "app/Components/FancyModal/FancyModalHeader"
-import { unsafe__getEnvironment } from "app/store/GlobalStore"
+import { unsafe__getEnvironment, unsafe_getFeatureFlag } from "app/store/GlobalStore"
```
Comment (nspinazz89) [inline src/app/Components/Bidding/Screens/BidResult.tsx:20]: out of curiosity why the double underscore on unsafe__getEnvironment, but only a single on unsafe_getFeatureFlag?

**Review by nspinazz89 — APPROVED** [review]
Looks good to me just a few questions

File: src/app/Components/Bidding/Screens/BidResult.tsx (hunk 3 above)
Comment (annacarey) [reply src/app/Components/Bidding/Screens/BidResult.tsx:20]: I have no idea! These are only for class components, so they're kind of deprecated hence the name 'unsafe'. That doesn't answer your question though :)

File: src/app/Components/Bidding/Screens/BidResult.tests.tsx (hunk 2 above)
Comment (annacarey) [reply src/app/Components/Bidding/Screens/BidResult.tests.tsx:216]: Oh yes you're totally right! It should say the sale artwork's end time is nil. Thank you for catching.

File: src/app/Components/Bidding/Screens/BidResult.tests.tsx [hunk 4]
```diff
@@ -195,6 +195,42 @@ describe("BidResult component", () => {
       })
     })
   })
+
+  describe("cascading end times", () => {
+    describe("with cascading end times turned on", () => {
```
Comment (mzikherman) [inline src/app/Components/Bidding/Screens/BidResult.tests.tsx:200]: Ah, ok, this is for the 'Cascading end time feature enabled'

**Review by mzikherman — APPROVED** [review]
Excellent!

**Pull request merged**